dashboard/test/integration/db_query_test.rb

```diff
@@ -20,7 +20,7 @@ def setup
       level: level,
       level_source: create(:level_source, level: level)
 
-    assert_cached_queries(11) do
+    assert_cached_queries(14) do
```

Comment:
This seems like a bit of a red flag. It looks like you have a couple checks to exit early in some of your methods e.g. if `family_name` is not present, but the change on this line seems to indicate that we're doing more queries even when no `family_name` is present.

Please make a best-effort attempt to eliminate these new queries if possible, or if not possible, please add a brief note here in PR comments why these new queries are difficult to avoid. 

If the source of the queries is not obvious, it might be easiest to find the right command to run from within bin/dashboard-console, which already logs SQL / ActiveRecord queries.

If that doesn't work, you might be able to turn on ActiveRecord / MySQL logging for your local application server here: https://github.com/code-dot-org/code-dot-org/blob/81c37d41c73f383ce4c6f92b28de9ffd6b25f98c/dashboard/config/environments/development.rb#L70-L71